### PR TITLE
Masasa_onlycreatebadge_permission

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
# Description
This PR creates a new Permission that restrict the badge management to capability to creating badges only. Below are pictures that show the task
Screenshot 2023-09-09 at 9 25 22 PM
Screenshot 2023-09-09 at 9 29 45 PM

## Related PRS (if any):
This frontend PR is related to PR [#534](https://github.com/OneCommunityGlobal/HGNRest/pull/534) on the backend
…

## Main changes explained:
- Created the permision in RolesTab.jsx and named it seeBadgeManagementTab …
- In order to have  to display its info icon description I added the permission in permissionList in Rolepermissions…
-  Then used the hasPermission function in badgemanagement to determine what to display between having the both the assign and develop badge tabs…
- The same function, hasPermission was also used in badgedevelopmentTable to disable the ability to edit or delete badges
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner.. this way you can assign and unassign the permission to you designated user...
4. go to otherlinks→ Permissions Management→ manage User permisssions→ Select user…
5. assign a user you can log into the permission 'See Badge Management Tab (ONLY create Badges)'
6. log out of owner and log in as that user 
7. go to otherlinks→ Badge Management

Expect to see this if the permission has been granted succesfully.. 
Screenshot 2023-09-09 at 9 43 44 PM

 repeat and log in as owner and remove the permission and expect to see this in the badge management tab:

Screenshot 2023-09-09 at 9 48 54 PM

## Note:
Include the information the reviewers need to know.
Do not test with an owner account because the owner always has all permissions

### CODE REVIEW

**Review by gabi-lima — COMMENTED** [review]
Hello! Even after saving the permission change, the other links tab does not appear in the account that I allowed viewing/editing.
image
Captura de tela 2023-09-10 015539

**Review by Haoj1 — COMMENTED** [review]
@ThapeloMasasa I have the same issue as @gabi-lima that I cannot see "other link" after giving the volunteer permission

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/f150a8c9-b240-4fa3-9430-f6e7f4f1b389

**Review by GaryB93 — DISMISSED** [review]
I'm also having the issue of not being able to see the Other Links tab for the user I assigned the permission to.
Screenshot 2023-09-12 at 1 34 20 PM

**Comment by ThapeloMasasa** [comment]
> I'm also having the issue of not being able to see the Other Links tab for the user I assigned the permission to. Screenshot 2023-09-12 at 1 34 20 PM

Hi @GaryB93 , I have added the backend code necessary. checkout into that branch. and it should work properly now..

**Review by navneeeth — DISMISSED** [review]
Hello, @ThapeloMasasa! I tested your PR and it works as expected. I logged in as an owner account and navigated to Other Links->Permission Management->Manage User Permissions->Select User and selected an user account that I can log into. To this user account, I added the 'See Badge Management Tab (ONLY create Badges)' permission and got the popup with the success message. Then after logging out of the owner account and logging in as the above user, I can see the 'Create New Badge' functionality available in Other Links->Badge Management. Good implementation!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/18c9b39d-7817-4375-a399-37cd59a897f1

**Comment by DavidC0126** [comment]
Hi @ThapeloMasasa, I tested your PR and here is what I found.
1. When I gave an Admin role permission to create badges, I was able to see the button and edit the badge.
2. If I gave a volunteer the permission, I cannot find other link to access the badge management page.

admin_with
admin_without
volunteer_with_per

**Comment by ThapeloMasasa** [comment]
> Hi @ThapeloMasasa, I tested your PR and here is what I found.
> 
> 1. When I gave an Admin role permission to create badges, I was able to see the button and edit the badge.
> 2. If I gave a volunteer the permission, I cannot find other link to access the badge management page.
> 
> admin_with admin_without volunteer_with_per

hi @DavidC0126 Please watch the demo video below.. also make sure you checkout out into the right backend branch and make sure to run npm install.. Thank you..


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/853e6d84-e9df-4073-8511-8ef7bf0b6c7c

**Review by Shiwani99 — DISMISSED** [review]
Hi @ThapeloMasasa,
I tested your PR and I'm having the same issue as mentioned previously, that is when I give the same badge permission to administrator, I'm able to get the same page as yours where I'm not able to edit or delete. When I set the same permission for a manager, I'm not able to find other links. I'm attaching a video of what I tested.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/e8b9feec-b0a4-4562-bf6e-2c766ec3dfb3

**Comment by GaryB93** [comment]
Hi @ThapeloMasasa, I pulled down the current backend and frontend for these PRs and was still unable to see the Other Links tab in the navigation. See video below.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/b4ed5716-2586-4c59-b160-3d511b537599

**Review by vikrambadhan — DISMISSED** [review]
Hi, I tested your PR and works as intended! Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/85417946/c41091eb-d283-443a-80fc-b34758d3ab09

**Comment by wantingxu7** [comment]
Hi, I tested the PR and after being assigned permission, I can see the badge management. But after removing the permission, I cannot find "other link" tab, neither badge management.
FE#1287 BE#534_after
FE#1287 BE#534_before

**Comment by TuanDinh140194** [comment]
Hi ! I have tested the PR and evverything look good for the role Admin as well as the role Manager. But I also don't see other link tab as Volunteer role like anyone else. However, I have found that if you change permission for volunteer role with create Badges enable. You can test it the same with other roles.  

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/81892917/9299733d-08f4-4a1d-930d-25d83581e201

](url)

**Comment by TuanDinh140194** [comment]
Here is the PR test for Volunteer Role

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/81892917/f79d59e2-ba3e-4250-a8dd-5d755cbf38b3

**Comment by TuanDinh140194** [comment]
Here is the test for Volunteer 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/81892917/f8975da5-42a4-423a-86a5-e037b3b76a47

Role

**Review by robertoooc — DISMISSED** [review]
Hi @ThapeloMasasa, I tested your branch and like most people after giving the create badge permission to a volunteer role I was not able to see the other links tab to lead to the badge management. I did however find a workaround by using the url route and confirmed that it does give the correct permission. I did create a new badge to confirm that this works and removing the permission does also restrict access to visiting the url route. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/2473179c-5a37-4df4-b07c-4a5feaf8cf52

**Review by TuanDinh140194 — DISMISSED** [review]

**Review by ChristianHrs — COMMENTED** [review]
Hi @ThapeloMasasa ,

I reviewed the code and it works well but only for users who already have access to the other links option. When giving permission to a volunteer account, I do not have access to the other links option.
Screenshot 2023-09-19 122647
Screenshot 2023-09-19 122342

**Review by GaryB93 — DISMISSED** [review]
Hi @ThapeloMasasa, I went ahead and reviewed this PR again trying to test any edge cases to see if I could access the Other Links tab. After giving both a Volunteer role and a Manager role the desired permission, I was still unable to access the Other Links tab.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/189c09f1-a2d8-4123-95b5-9e8dadd9f568

**Review by iTvX — COMMENTED** [review]
I followed all your instructions, but I cannot see the other link than in my other account after adding permission. That's so weird.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/b84106c8-1c9c-4c5a-87a5-60c36e2a452e

**Review by StrawberryCalpico — APPROVED** [review]
I tested your PR. It works as intended! good implementation! 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114443664/208fa59d-c9d6-4f2f-a0ca-2277fafa45ed

**Review by malikjahanzaib — APPROVED** [review]
Hi @ThapeloMasasa, I have reviewed this PR and the implemented functionality works as intended. Here is my test recording below. Great job!!!

Test rec:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46221334/7f4f8aaa-9e30-403d-8eeb-ca28892f7e05

**Comment by RheaWu1212** [comment]
Hi I'm testing your PR, it's able to assign 'create badge' and the button showed up but no existing list below:

Screen Shot 2023-10-14 at 20 15 39
Screen Shot 2023-10-14 at 20 21 22

**Comment by ThapeloMasasa** [comment]
> Hi I'm testing your PR, it's able to assign 'create badge' and the button showed up but no existing list below:
> 
> Screen Shot 2023-10-14 at 20 15 39 Screen Shot 2023-10-14 at 20 21 22

Hi @RheaWu1212, I am not sure I understand what probelm you are exeperiencing here.. please clarify so that we can know if it is related to this PR.

**Comment by ThapeloMasasa** [comment]
> Hi @ThapeloMasasa, I pulled down the current backend and frontend for these PRs and was still unable to see the Other Links tab in the navigation. See video below.
> 
>  Screen.Recording.2023-09-13.at.2.02.33.PM.mov

Hi @GaryB93, please  check the demo below.. that is how a volunteer behaves without  and  with the permsission. 

Uploading Screen Recording 2023-10-24 at 5.44.50 PM.mov…

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -56,6 +56,8 @@ export const Header = props => {
   const canPutUserProfileImportantInfo = props.hasPermission('putUserProfileImportantInfo');
   // Badges
   const canCreateBadges = props.hasPermission('createBadges');
+  const canOnlyCreateBadges = props.hasPermission('seeBadgeManagementTab');
+  console.log('canOnlyCreateBadges', canOnlyCreateBadges)
```
Comment (nathanah) [inline src/components/Header/Header.jsx:60]:
```suggestion
```

**Review by nathanah — CHANGES_REQUESTED** [review]

**Review by shubhankarval — APPROVED** [review]
Hey @ThapeloMasasa, the PR works as intended, great job!

When user has access to the Badge Management Tab:
Screenshot 2023-11-04 171038

When user does not have access to the Badge Management Tab:
Screenshot 2023-11-04 171300

**Review by Alforoan — APPROVED** [review]
It works. Nice!
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/704e065c-140b-4f74-82d1-64d11ddd2252

**Review by Pandani07 — DISMISSED** [review]
Tested this PR and the backend PR, the functionality works as intended. I added the permission for my volunteer account and I could see badge management under 'other links'.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/a0e62373-b7f7-4b18-a950-b1fcca3871ad

**Review by skaush21 — COMMENTED** [review]
Tested createbadge permission

* logged in as owner
* when signed in received a bunch of error 400's (might not be related to PR)
* navigated to user permissions -> user -> added create badge permission
* logged out
* logged in as volunteer user
* verified that user is able to see badges
* logged out
* logged in as owner, removed user permission
* logged out and back into volunteer account
* volunteer is no longer able to see bages.
feature works great


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/141872402/8fb7c014-68c5-4ad5-8dae-5c343b62d0f8
